### PR TITLE
[tests/lag_2] Convert single_lag_test.yml to pytest

### PR DESCRIPTION
**Pull request by Blueve**
Convert single_lag_test.yml to pytest

1. Convert single_lag_test.yml to pytest
2. Convert lag_minlink.yml to pytest

I will use the pytest version lag test replace the original lag_2.yml after all sub-testcase converted.

### Description of PR

Summary:
Fixes # (issue)

### Type of change

- [ ] Bug fix
- [ ] Testbed and Framework(new/improvement)
- [x] Test case(new/improvement)

#### How did you do it?
1. Convert single_lag_test.yml to pytest
2. Convert lag_minlink.yml to pytest
3. Use ansible eos config module to shut/noshut interface

#### How did you verify/test it?
Run test in local vstestbed
```
=============================== warnings summary ===============================
test_lag_2.py::test_lag_2
  /usr/local/lib/python2.7/dist-packages/pytest_ansible/module_dispatcher/v28.py:70: UserWarning: provided hosts list is empty, only localhost is available
    warnings.warn("provided hosts list is empty, only localhost is available")

-- Docs: https://docs.pytest.org/en/latest/warnings.html
=================== 1 passed, 1 warnings in 1021.49 seconds ====================
```

### CODE REVIEW

File: tests/test_lag_2.py [hunk 1]
```diff
@@ -92,6 +98,26 @@ def check_single_lag_lacp_rate(common_setup_teardown, nbrhosts, lag_name):
...
 
+def check_single_lag(common_setup_teardown, nbrhosts, lag_name):
+    duthost, ptfhost, testbed, vm_neighbors, mg_facts, lag_facts, fanout_neighbors = common_setup_teardown
+    logging.info("Start checking single lap for: %s" % lag_name)
+
+    intf, po_interfaces = get_lag_intfs(lag_facts, lag_name)
+    po_flap = check_flap(lag_facts, lag_name)
+
+    # Figure out fanout switches info if exists for the lag member and run minlink test
+    if intf in fanout_neighbors.keys():
```
Comment (Blueve) [inline tests/test_lag_2.py]: There are no fanout switch in vstestbed, then I add this check to skip this test

File: tests/test_lag_2.py [hunk 2]
```diff
@@ -106,6 +132,72 @@ def verify_lag_lacp_timing(ptfhost, vm_name, lacp_timer, exp_iface):
...
+        # Refresh lag facts
+        lag_facts = duthost.lag_facts(host = duthost.hostname)['ansible_facts']['lag_facts']
+
+        # Verify lag member is marked deselected for the shutdown porta and all other lag member interfaces are marked selected
+        for po_intf in po_interfaces.keys():
+            logging.info("po: %s | %s | %s" % (po_intf, intf, lag_facts['lags'][lag_name]['po_stats']['ports'][po_intf]['runner']['selected']))
+            if po_intf != intf:
+                assert lag_facts['lags'][lag_name]['po_stats']['ports'][po_intf]['runner']['selected']
+            else:
+                assert not lag_facts['lags'][lag_name]['po_stats']['ports'][po_intf]['runner']['selected']
```
Comment (Blueve) [inline tests/test_lag_2.py]: I got assertion fail in my vstestbed...
No clue by now

File: tests/test_lag_2.py [hunk 3]
```diff
@@ -119,8 +211,38 @@ def get_conn_graph_facts(testbed_devices, host):
...
\ No newline at end of file
+    logging.info("Set interface [%s] lacp rate to [%s]" % (intf, mode))
+
+def set_neighbor_interface(vm_host, neighbor_interface, shut):
+    vm_host.eos_config(
+        lins=['%sshutdown' % ('' if shut else 'no ')],
+        parents='interface %s' % neighbor_interface)
+    logging.info('%s interface [%s]' % ('Shut' if shut else 'No shut', neighbor_interface))
+
+def shell_retry(host, command, retries, delay):
```
Comment (Blueve) [inline tests/test_lag_2.py]: I am not sure if pytest ansible support retries naturally, then I create this method by myself.
If we need create a more general retry util, we might need allow user to pass a lambda expr to this method to check the output result.

File: tests/test_lag_2.py [hunk 4]
```diff
@@ -106,6 +132,72 @@ def verify_lag_lacp_timing(ptfhost, vm_name, lacp_timer, exp_iface):
...
+
+        # Refresh lag facts
+        lag_facts = duthost.lag_facts(host = duthost.hostname)['ansible_facts']['lag_facts']
+        for po_intf in po_interfaces.keys():
+            assert lag_facts['lags'][lag_name]['po_stats']['ports'][po_intf]['runner']['selected'] == True
+        
+        assert lag_facts['lags'][lag_name]['po_intf_stat'] == 'Up'
+
 @pytest.fixture(scope="module")
 def conn_graph_facts(testbed_devices):
```
Comment (yxieca) [inline tests/test_lag_2.py]: Please use https://github.com/Azure/sonic-mgmt/blob/master/tests/common/fixtures/conn_graph_facts.py#L5 instead.

Example here: https://github.com/Azure/sonic-mgmt/pull/1503/files#diff-f69c1b6dd64cba3c159d7f379026a950R20

File: tests/test_lag_2.py (hunk 3 above)
Comment (yxieca) [reply tests/test_lag_2.py]: There is a wait_until helper function, maybe you can take a look to see if that addresses your needs?

File: tests/test_lag_2.py (hunk 2 above)
Comment (yxieca) [reply tests/test_lag_2.py]: add debug log before assert?

Comment (Blueve) [reply tests/test_lag_2.py]: > add debug log before assert?

I plan to let the test sleep more time and check the interface's status in eos vm directly

Comment (Blueve) [reply tests/test_lag_2.py]: This is caused by a typo which cause the ansible command execution fail. I've fixed it

File: tests/test_lag_2.py [hunk 5]
```diff
@@ -106,21 +133,88 @@ def verify_lag_lacp_timing(ptfhost, vm_name, lacp_timer, exp_iface):
...
\ No newline at end of file
+    logging.info("Set interface [%s] lacp rate to [%s]" % (intf, mode))
+
+def set_neighbor_interface(vm_host, neighbor_interface, shut):
+    vm_host.eos_config(
+        lines=['%sshutdown' % ('' if shut else 'no ')],
+        parents='interface %s' % neighbor_interface)
+    logging.info('%s interface [%s]' % ('Shut' if shut else 'No shut', neighbor_interface))
+
+def check_shell_output(host, command):
```
Comment (yxieca) [inline tests/test_lag_2.py]: This can be a shared helper function.

File: tests/test_lag_2.py [hunk 6]
```diff
@@ -106,21 +133,88 @@ def verify_lag_lacp_timing(ptfhost, vm_name, lacp_timer, exp_iface):
...
 
 def set_interface_lacp_rate(vm_host, intf, mode):
     vm_host.eos_config(
         lines=['lacp rate %s' % mode],
         parents='interface %s' % intf)
-    logging.info("Set interface [%s] lacp rate to [%s]" % (intf, mode))
\ No newline at end of file
+    logging.info("Set interface [%s] lacp rate to [%s]" % (intf, mode))
+
+def set_neighbor_interface(vm_host, neighbor_interface, shut):
```
Comment (yxieca) [inline tests/test_lag_2.py]: Does it make sense to make this a method of vm_host?

Comment (Blueve) [reply tests/test_lag_2.py:212]: Yes. I can move this to EosHost in next PR

File: tests/test_lag_2.py [hunk 7]
```diff
@@ -92,6 +99,26 @@ def check_single_lag_lacp_rate(common_setup_teardown, nbrhosts, lag_name):
...
+    logging.info("Start checking single lap for: %s" % lag_name)
+
+    intf, po_interfaces = get_lag_intfs(lag_facts, lag_name)
+    po_flap = check_flap(lag_facts, lag_name)
+
+    # Figure out fanout switches info if exists for the lag member and run minlink test
+    if intf in fanout_neighbors.keys():
+        peer_device = fanout_neighbors[intf]['peerdevice']
+        neighbor_interface = fanout_neighbors[intf]['peerport']
+        vm_host = nbrhosts[peer_device]
```
Comment (Blueve) [inline tests/test_lag_2.py:113]: Seems like the nbrhosts are not include fanout swtichs, I need to find a way to access and operate fanout switch

**Pull request merged**